### ISSUE TITLE
New files created but no PR is made

### ISSUE THREAD

**Issue by thieman** [peter-evans/create-pull-request#59]
I've created a new file, but when I run the action all it outputs is "Repository has no modified or untracked files. Skipping.". I've confirmed that right before I run create-pull-request, a `git status` shows my new file. I'm not sure what's breaking since the script isn't logging anything other than the skipping notification.

I suspect there's an issue with the logic here: https://github.com/peter-evans/create-pull-request/blob/96ac64ddca6433d9deb7799065903029788480dc/create-pull-request.py#L71-L75

It looks like if we hit this `except:` block that we'll never recover our changes from the stash? I'm not a git expert so my understanding here may be wrong, but seems like we can't recover correctly in this case. I'll keep digging in and let you know if I figure out what's going on.

edit: yeah, this is definitely what's wrong. Got some output from that except handler:

```
Exception in checkout_branch
Cmd('git') failed due to: exit code(1)
  cmdline: git stash pop
  stdout: 'The stash entry is kept in case you need it again.'
  stderr: '.github/workflows/update-owning-team.yml already exists, no checkout
error: could not restore untracked files from stash'
```

edit2: Okay, I think I understand the problem. It looks like a previous run of the action successfully created the branch but errored out before it could make a PR. So the branch is there, I just can't see it. The action then (perhaps correctly) bails out without doing anything since the branch is up-to-date.

Maybe we need to check on the PR status directly?

**Comment by peter-evans** [comment]
Thank you for raising this issue. I would really like to understand your use-case here and try and reproduce the problem. If you could provide more detail that would be great. e.g. (create-pull-request version, your entire workflow, your use-case, etc.) If it's in open source a link to the repository would be even better.

Just trying to guess based on the error message you provided. Are you trying to modify a workflow file during the execution of a workflow? There are definitely restrictions with this. You mentioned that the first run failed. What was the error message then?

If you try and modify workflow files and push them during the execution of a workflow you might see an error message like this:
```
git.exc.GitCommandError: Cmd('git') failed due to: exit code(1)
  cmdline: git push -f --set-upstream origin test-patches
  stderr: 'To https://github.com/peter-evans/actions-playground
 ! [remote rejected] test-patches -> test-patches (refusing to allow an integration to create or update .github/workflows/create-pull-request.yml)
```

To get to the bottom of this issue I think we need to focus on the use-case you have and the error thrown the *first time* the action ran.

**Comment by thieman** [comment]
I think this is what happened:

1. I ran the action from a dummy, non-master branch. This opened a PR against the non-master branch (intended behavior)
2. I then deleted the dummy branch since I was just pushing to it to trigger an action build and it did not have any actual changes in it
3. GitHub automatically closed the PRs before I even saw them, since they were based on a now-deleted branch

So what I observed was "there are no PRs, and running the action again doesn't make PRs" but in reality I think everything did what it should have done.

I think what I _wanted_ to do was pass `PULL_REQUEST_BASE=master` in step 1 so that the PR got opened against master, not my dummy branch.

I think the only change I might ask for from this issue would be for the action to be a bit more verbose in logging what's going on. I threw a bunch of print statements in my fork of the action which is what helped me track down what was going on.

I'll close this issue, thanks!

**Comment by peter-evans** [comment]
Glad you figured it out. You're right, the logging could be much better. I will look into adding some more verbose and understandable messages.

**Comment by danopia** [comment]
I ran into another case where this error message seemed wrong, in my case because there was already a PR open from a previous run which (sensibly) made the same change. I eventually realized this message was telling me that, _based on the existing PR_ there were no new changes to commit. Perhaps in this case a message more like `Repository has no modified or untracked files compared to existing PR. Skipping.` would be more clear.
